### ISSUE TITLE
Does Tensort RT work with SDXL-medvram option?

### ISSUE THREAD

**Issue by aggregate15** [NVIDIA/Stable-Diffusion-WebUI-TensorRT#106]
So I have Tensor RT working with SD1.5 at almost 2X the speed on my 8GB VRAM Card, Thank you Nvidia for this great  improvement. 

With my card I use Medvram option for SDXL. I have also created SDXL Profiles on a dev environment . Normally the SDXL models work fine using medvram option, taking around 2 it/s, but when i use Tensor RT profile for SDXL, it seems like the medvram option is not being used anymore as the iterations start taking several minutes as if the medvram option is disabled. I have trained profiles using both medvram options enabled and disabled but the iterations are still seemingly not using medvram. Is Tensor RT not configured to use medvram ? And if not then is there a workround or a command to make it use medvram option with Tensor RT ?

**Comment by boricuapab** [comment]
I've noticed the same thing, it's actually slower to generate images using an sdxl tensor rt converted model.  I have an rtx 2060 super 8gb gpu

![dynavisXL1024pxTRT0003](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/61924816/22a1a119-a706-449c-808f-77eb8fb182ab)

![dynavisXL1024pxTRT0004](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/61924816/a213b283-e3ac-46da-b5a4-4cb37076c904)

![dynavisXL1024pxTRT0005](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/61924816/9dd738da-1206-4052-9c8b-d578ca550ae8)

**Comment by boricuapab** [comment]
Using the original dynavis xl model I get generations at 2s/it (without --med vram)

![dynavisXL1024px0008](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/61924816/84dd7069-afba-4067-b4b3-0a79baa97902)

![dynavisXL1024px0006](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/61924816/5b25fa44-a08e-4dba-998e-bc8c2256be75)

![dynavisXL1024px0007](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/61924816/d1519e06-4889-4334-afe9-7c7a52db6e2f)

**Comment by boricuapab** [comment]
Hmmm, it does get faster on the second generation using the trt converted sdxl model albeit still slower than the default model

![dynavisXL1024pxTRT0006](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/61924816/295b1f93-1cae-4c12-9209-0b6163d70e28)

![dynavisXL1024pxTRT0007](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/61924816/bb68a7f3-113c-4692-b403-6bc3ba8c3a4f)

![dynavisXL1024pxTRT0008](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/61924816/0d83ee8c-8d1d-4304-ad33-59bbeff0e087)

**Comment by aggregate15** [comment]
Yeah reading through some other comments, it seems like its not really optimized for SDXL. Its a dev feature after all and maybe in the actual patch there will be a more stable option. 

Also if you look at NVIDIA Articles, seems like Tensor RT gives more improvements at 512X512 and lower improvements on 768X768. So maybe, as the resolution Increases to 1024X1024 the returns are not as good. Anyway, even the SD1.5 is a huge perk that came out of nowhere from Nvidia, so I'm happy enough with even that, as it is.

**Comment by trentaiq** [comment]
![image](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/127002262/e9a5217a-5322-4c96-b7b5-5c38be1bc617)
I found lowvram to work very fast, generating in as low as 5 seconds at 1024x1024, and got the same very slow results with medvram

**Comment by wizz13150** [comment]
+1

`--lowvram` is the way for 12gb or less ! 
Same speed than without the arg, no impact on inference speed, using trt.

1024*1024 with a rtx3060 - 12gb:
![image](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/22177081/58ae1c94-dca3-449a-a6b1-ad7e5a6ec30c)

**Comment by boricuapab** [comment]
Indeed using --lowvram flag we get faster iterations speeds, hitting 2it/s

![dynavisXL1024pxLowVramTRT0009](https://github.com/NVIDIA/Stable-Diffusion-WebUI-TensorRT/assets/61924816/a3d15174-e2b8-42b0-8c8a-d9a2c4c96374)

**Comment by wizz13150** [comment]
Hey, update about this :

https://nvidia.custhelp.com/app/answers/detail/a_id/5490

**Comment by boricuapab** [comment]
> Hey, update about this :
> 
> https://nvidia.custhelp.com/app/answers/detail/a_id/5490
> 
> Don't need `--lowvram` anymore !
> 
> Cheers ! 🥂

Tested this on RTX 2060 super 8gb card and it OOM, so for now I'll keep using the low vram flag.

**Comment by LurkingLearner** [comment]
Even on an RTX 4090 I am having abysmal generation times when trying to run TRT and an SDXL model - sometimes an order of magnitude slower. This is while running the latest dev branch of both A1111 and TRT.

SD models I am seeing the expected gains (~80% faster at 768x768). So, for whatever it's worth, this problem seems to be present even with "plenty" of VRAM available.
